### ISSUE TITLE
httpsRequestPromisified throws [Object object]

### ISSUE THREAD

**Issue by tombuckley91** [skyhookadventure/appsync-client#2]
Firstly, thank you very much for this client, I really did not like aws-appsync and was having a heap of problems with it and esbuild, so I appreciate this significantly slimmer package!

Just whilst I was doing some testing I came across a defect with the way errors are thrown, line 19 of httpsRequestPromisified throws the parsed object, which, stringified, results in [Object object]. I have put a branch together for this and updated the tests.

Thank you again,
Tom

**Comment by tombuckley91** [comment]
@alan-cooney  - Evening bud, you had chance to look at this?

**Comment by alan-cooney** [comment]
Hi @tombuckley91 - thanks for the PR & follow up. I've merged your PR so it will filter through to npmjs automatically in a few minutes.

By the way, we've also found similar problems with the AWS client. However if you do find yourself needing more advance Apollo client features, you can get it to work by using `aws-appsync-auth-link` directly. Here's a snippet from our React codebase (I've removed some Skyhook specific code so shout if it doesn't work):

```typescript
import { createAuthLink, AuthOptions } from 'aws-appsync-auth-link';
import { createSubscriptionHandshakeLink } from 'aws-appsync-subscription-link';
import {
  ApolloClient,
  from,
  HttpLink,
  InMemoryCache,
  split,
  NormalizedCacheObject,
} from '@apollo/client';

/**
 * Create Appsync Client
 *
 * Similar to AWSAppSyncClient from 'aws-appsync', but works with Apollo 3 rather than <2.6. Based on
 * https://github.com/awslabs/aws-mobile-appsync-sdk-js/pull/561#issuecomment-701696316
 */
export function createAppsyncClient({
  url,
  region,
  auth,
  name,
}: AppsyncClientConfig): ApolloClient<NormalizedCacheObject> {
  const httpLink = new HttpLink({ uri: url });

  return new ApolloClient({
    name,
    cache: new InMemoryCache(),
    link: from([
      createAuthLink({ auth, url, region }),
      split(
        (op) => {
          const { operation } = op.query.definitions[0] as any;
          return operation !== 'subscription';
        },
        httpLink,
        createSubscriptionHandshakeLink({ auth, url, region }, httpLink),
      ),
    ]),
  });
}

export interface AppsyncClientConfig {
  /** Client name */
  name: string;
  /** Appsync API URL */
  url: string;
  /** Appsync AWS Region (e.g. eu-west-1) */
  region: string;
  /** Auth configuration */
  auth?: AuthOptions;
}
```

**Comment by tombuckley91** [comment]
@alan-cooney - Good morning Alan, god this is something I would have loved to have seen about a month ago! I've just finished setting up authenticated, filtered subscriptions against a user pool, which ended up in basically an identical implementation to what you've used above. I really wanted to use Apollo as well and not the amplify client, but lord knows I had to pay for that decision haha. I use your client in my lambdas, I love that it's very basic, it's all I need it to do. I'll put the PR together now for the HTTPS events, I've been using a fork of it for a week and it's been grand. Cheers, Tom.
